### PR TITLE
show-midi: init at 0.8.0

### PR DESCRIPTION
**Pull request by minijackson**
## Description of changes

Really nice tool to show MIDI events: https://github.com/gbevin/ShowMIDI

ShowMIDI is compatible with macOS, but I don't have one so for now this package is Linux only. Help for a macOS port would be appreciated.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Tested basic functionality of LV2 and VST3 plugins
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sh/show-midi/package.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    libXinerama
+    libXext
+    libXcursor
+  ];
+
+  enableParallelBuilding = true;
+
+  makeFlags = [
+    "-C"
+    "Builds/LinuxMakefile"
```
Comment (AndersonTorres) [inline pkgs/by-name/sh/show-midi/package.nix:41]:
```suggestion
    "-C"  "Builds/LinuxMakefile"
```

P.S.: can't you try to set `makefile = Builds/LinuxMakefile`?

File: pkgs/by-name/sh/show-midi/package.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    "VST3_MANIFEST_HELPER"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dt $out/share/ShowMIDI/themes Themes/*
+
+    mkdir -p $out/bin $out/lib/lv2 $out/lib/vst3
+    cd Builds/LinuxMakefile/build/
```
Comment (AndersonTorres) [inline pkgs/by-name/sh/show-midi/package.nix:58]: pushd and popd are better for this.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sh/show-midi/package.nix (hunk 1 above)
Comment (minijackson) [reply pkgs/by-name/sh/show-midi/package.nix:41]: `"-C" "Builds/LinuxMakefile"` doesn't work well with the auto-formatter, but I can do `"-C Builds/LinuxMakefile"`.

I did try `makefile = "..."`, but it doesn't change the working directory, which is needed.

File: pkgs/by-name/sh/show-midi/package.nix (hunk 2 above)
Comment (minijackson) [reply pkgs/by-name/sh/show-midi/package.nix:58]: I can use `pushd`, and `popd`, but with the way I've written it, I wouldn't need `popd`, so I think `cd` is clearer.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/sh/show-midi/package.nix (hunk 2 above)
Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/sh/show-midi/package.nix:57]: +1 to pushd / popd or `cd Builds/LinuxMakefile/build/ ... cd -`. most other packages don't  exit installPhase in a different directory than it started in.

**Review by ghost — COMMENTED** [review]
Result of `nixpkgs-review pr 277901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>show-midi</li>
  </ul>
</details>

**Pull request merged**